src/System.Management.Automation/engine/runtime/Operations/MiscOps.cs

```diff
@@ -1603,33 +1603,38 @@ internal static void CheckActionPreference(FunctionContext funcContext, Exceptio
             // set $? to false indicating an error
             context.QuestionMarkVariableValue = false;
 
-            bool anyTrapHandlers = funcContext._traps.Any() && funcContext._traps.Last().Item2 != null;
+            ActionPreference preference = ExceptionHandlingOps.QueryForAction(rte, rte.Message, context);
 
-            if (!anyTrapHandlers && !ExceptionHandlingOps.NeedToQueryForActionPreference(rte, context))
+            // if the exception is new (i.e. it was not rethrown from a naked throw
+            // statement inside of a catch block, and it is not currently being
+            // handled by the system), and if ErrorActionPreference is set to break,
+            // then break immediately into the debugger.
+            if (!rte.WasRethrown &&
+                context.CurrentExceptionBeingHandled == null &&
+                preference == ActionPreference.Break)
             {
-                throw rte;
+                context.Debugger?.Break(rte);
+
+                // set the value of $? here in case it was reset in the debugger.
+                context.QuestionMarkVariableValue = false;
             }
 
-            ActionPreference preference;
+            bool anyTrapHandlers = funcContext._traps.Any() && funcContext._traps.Last().Item2 != null;
```

Comment:
Can we avoid LINQ?
What is "Item2" here?

Reply:
No idea what Item2 is. That was existing code that I just moved. I'll have to take a closer look, and see if LINQ can be removed as well.

Reply:
Removed this use of LINQ and added comment describing what Item2 is.